pkg/controller/astarte/reconcile/utils.go

```diff
@@ -390,9 +390,14 @@ func getImageForClusteredResource(defaultImageName, defaultImageTag string, reso
 	return image
 }
 
-func getDeploymentStrategyForClusteredResource(cr *apiv1alpha1.Astarte, resource apiv1alpha1.AstarteGenericClusteredResource) appsv1.DeploymentStrategy {
+func getDeploymentStrategyForClusteredResource(cr *apiv1alpha1.Astarte, resource apiv1alpha1.AstarteGenericClusteredResource, component apiv1alpha1.AstarteComponent) *appsv1.DeploymentStrategy {
```

Comment:
Nope. Don't return a pointer, you're overcomplicating the code. Keep the pointer math hidden behind this function.